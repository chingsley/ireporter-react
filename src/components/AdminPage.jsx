import React from 'react';
import ReportRow from './TableRow';

const report1 = {
  address: 'Winterfell',
  coordinates: '2392, 298202',
  comment: 'We want to report corruption',
  images: '',
  videos: '',
  status: 'draft',
  type: 'red-flag',
  createdon: Date.now(),
  createdby: 'James'
};
const report2 = {
  address: 'Volaris',
  coordinates: '02902939, 248230',
  comment: 'We need some help here',
  images: '',
  videos: '',
  status: 'draft',
  type: 'red-flag',
  createdon: Date.now(),
  createdby: 'John'
};

const AdminPage = () => (
  <div className="row-1-of-3" id="table-wrapper">
    <table className="table-admin" id="table-admin">
      <tbody>
        <tr className="row">
          <td className="cell">ID</td>
          <td className="cell">Type</td>
          <td className="cell">Comment / Media</td>
          <td className="cell">Location</td>
          <td className="cell">Created on</td>
          <td className="cell">Created by</td>
          <td className="cell">Status</td>
        </tr>
        <ReportRow report={report1} />
        <ReportRow report={report2} />
      </tbody>
    </table>
  </div>
);

export default AdminPage;
