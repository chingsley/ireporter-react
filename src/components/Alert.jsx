import React, { Component, Fragment } from 'react';
import styled from 'styled-components';
import EventEmitter from 'event-emitter';

const Container = styled.div`
  background-color: ${props => props.bgColor};
  color: white;
  padding: 16px;
  position: fixed;
  top: ${props => props.top}px;
  right: 100px;
  z-index: 999;
  transition: top 0.5s ease;

  > i {
    margin-left: 8px;
  }
`;

const emitter = new EventEmitter();

export const alertUser = (msg, type) => {
  emitter.emit('notification', msg, type);
};
class Alert extends Component {
  constructor(props) {
    super(props);

    this.state = {
      top: -100,
      msg: '',
      type: '',
    };

    this.timeout = null;

    emitter.on('notification', (msg, type) => {
      this.onShow(msg, type);
    });
  }

  onShow = (msg, type) => {
    if (this.timeout) {
      clearTimeout(this.timeout);
      this.setState({ top: -100 }, () => {
        this.timeout = setTimeout(() => {
          this.showAlert(msg, type);
        }, 500);
      });
    } else {
      this.showAlert(msg, type);
    }
  }

  showAlert = (msg, type) => {
    this.setState({ top: 200, msg, type }, () => {
      this.timeout = setTimeout(() => {
        this.setState({ top: -100 });
      }, 5000);
    });
  }

  render() {
    const { top, type, msg } = this.state;
    return (
      <Fragment>
        <Container top={top} bgColor={type === 'error' ? 'red' : 'green'}>
          {msg}
        </Container>
      </Fragment>
    );
  }
}

export default Alert;
