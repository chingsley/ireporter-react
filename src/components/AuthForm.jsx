import React, { Fragment, Component } from 'react';

export default class SignUpForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      firstname: '',
      lastname: '',
      username: '',
      othernames: '',
      email: '',
      password: '',
      phoneNumber: '',
      adminSecret: ''
    };

    this.onSubmit = this.onSubmit.bind(this);
    this.onFirstnameChange = this.onFirstnameChange.bind(this);
  }

  onSubmit(e) {
    e.preventDefault();
    const { onSubmit } = this.props;
    onSubmit({ ...this.state });
  }

  onFirstnameChange(e) {
    const firstname = e.target.value;
    this.setState({ firstname });
  }

  render() {
    const {
      firstname,
      lastname,
      othernames,
      username,
      email,
      password,
      phoneNumber,
      adminSecret
    } = this.state;
    const { formType } = this.props;
    return (
      <form onSubmit={this.onSubmit} className="form">
        {formType === 'registration' && (
          <Fragment>
            <input
              type="text"
              id="firstname"
              value={firstname}
              onChange={this.onFirstnameChange}
              placeholder="first name"
              required
            />
            <input
              type="text"
              id="lastname"
              value={lastname}
              onChange={e => this.setState({ lastname: e.target.value })}
              placeholder="last name"
              required
            />
            <input
              type="text"
              id="othernames"
              value={othernames}
              onChange={e => this.setState({ othernames: e.target.value })}
              placeholder="other names"
            />
            <input
              type="text"
              id="username"
              value={username}
              onChange={e => this.setState({ username: e.target.value })}
              placeholder="username"
            />
            <input
              type="text"
              id="phonenumber"
              placeholder="phone number"
              value={phoneNumber}
              onChange={e => this.setState({ phoneNumber: e.target.value })}
              required
            />
            <input
              type="text"
              id="adminSecret"
              placeholder="admin secret key"
              value={adminSecret}
              onChange={e => this.setState({ adminSecret: e.target.value })}
            />
            <input type="file" id="input-file-pic" />
          </Fragment>
        )}
        <input
          type="text"
          id="email"
          placeholder="email"
          value={email}
          onChange={e => this.setState({ email: e.target.value })}
          required
        />
        <input
          type="text"
          id="password"
          placeholder="password"
          value={password}
          onChange={e => this.setState({ password: e.target.value })}
          required
        />
        <button className="btn-reg btn" id="btn-reg">
          { formType === 'registration' ? 'REGISTER' : 'LOGIN' }
        </button>
      </form>
    );
  }
}
