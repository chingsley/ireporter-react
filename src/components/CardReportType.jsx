import React, { Fragment } from 'react';

const CardReportType = () => (
  <Fragment>
    <div className="card">
      <div className="card-picture card-picture--2">&nbsp;</div>
      <h4 className="card-heading">
        <span className="card-heading-span card-heading-span--2">
          Intervention Records
        </span>
      </h4>
      <div className="card-details">
        <ul>
          <li>Bad road sections</li>
          <li>Flooding</li>
          <li>Communal violence</li>
          <li>Disease outbreak</li>
          <li>... and many more</li>
        </ul>
      </div>
      <div className="link-container u-center-item">
        <a href="report.html" className="link report-a-case">
          Report a case &rarr;
        </a>
      </div>
    </div>
  </Fragment>
);

export default CardReportType;
