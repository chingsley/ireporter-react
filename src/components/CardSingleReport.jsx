import React from 'react';

const CardSingleReport = () => (
  <div className="report" id="report-cards">
    <button className="btn-report-delete">&times;</button>
    <div className="div-report-img">
      <img className="report-img" src="img/1543731908425.jpg" alt="alt" />
    </div>
    <div className="report-details">
      <table className="report-details-table">
        <tr>
          <td>Type :</td>
          <td>Intervention</td>
        </tr>
        <tr>
          <td>Created On :</td>
          <td>12 Nov, 2018</td>
        </tr>
        <tr>
          <td>Details :</td>
          <td className="animated-link">
            <a
              href="#popup"
              className="report-card-comments-and-media"
            >
              ...
            </a>
          </td>
        </tr>
        <tr>
          <td>Status :</td>
          <td className="td-status-draft">Draft</td>
        </tr>
      </table>
    </div>
    <div className="report-details-div-btn">
      <button className="btn" id="btn-edit-report">
        Edit
      </button>
    </div>
  </div>
);

export default CardSingleReport;
