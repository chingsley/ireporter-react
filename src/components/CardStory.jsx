/* eslint-disable max-len */
import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const defaultText = 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Dignissimos, facere at. Deserunt, accusantium tempora similique veniam sed sapiente consequatur adipisci asperiores ratione illo molestiae animi, voluptate quam ipsa dignissimos aliquid?';
const CardStory = ({ src, title, text }) => (
  <div className="card">
    <div className="card__img-container">
      <img src={src} alt="Smiley face" className="card__img" />
    </div>
    <div className="card__text-container">
      <p>
        <h2><Link className="link-on-black" to="/">{title}</Link></h2>
        {text}
      </p>
    </div>
  </div>
);

CardStory.propTypes = {
  src: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  text: PropTypes.string,
};

CardStory.defaultProps = {
  text: defaultText,
};

export default CardStory;
