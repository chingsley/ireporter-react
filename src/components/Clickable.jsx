/* eslint-disable react/jsx-one-expression-per-line */
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Clickable = ({ className, text }) => (
  <div className={`link-container ${className}`}>
    <Link to="/" className="link">{text}</Link>
  </div>
);

Clickable.propTypes = {
  className: PropTypes.string,
  text: PropTypes.string.isRequired,
};

Clickable.defaultProps = {
  className: '',
};

export default Clickable;
