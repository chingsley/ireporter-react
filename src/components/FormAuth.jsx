import React, { Fragment, Component } from 'react';
import propTypes from 'prop-types';
import { connect } from 'react-redux';
import { authorise } from '../actions/auth';

export class FormAuth extends Component {
  constructor(props) {
    super(props);

    this.state = {
      firstname: '',
      lastname: '',
      username: '',
      othernames: '',
      email: '',
      password: '',
      phoneNumber: '',
      adminSecret: '',
    };

    this.formType = props.formType;
    this.history = props.history;

    this.onSubmit = this.onSubmit.bind(this);
  }

  async onSubmit(e) {
    e.preventDefault();
    const {
      authorise: handleAuthorisation,
    } = this.props;
    await handleAuthorisation({ ...this.state }, this.formType);

    const { isLoggedIn, fetching } = this.props;
    if (!fetching && isLoggedIn) {
      this.history.push('/');
    }
  }

  render() {
    const {
      firstname,
      lastname,
      othernames,
      username,
      email,
      password,
      phoneNumber,
      adminSecret,
    } = this.state;
    return (
      <div className="container">
        <form onSubmit={this.onSubmit} className="container__form">
          <input
            className="container__form-input"
            type="email"
            id="email"
            placeholder="email"
            value={email}
            onChange={e => this.setState({ email: e.target.value })}
            required
          />
          <input
            className="container__form-input"
            type="text"
            id="password"
            placeholder="password"
            value={password}
            onChange={e => this.setState({ password: e.target.value })}
            required
          />
          {this.formType === 'signup' && (
            <Fragment>
              <input
                className="container__form-input"
                type="text"
                id="firstname"
                value={firstname}
                onChange={e => this.setState({ firstname: e.target.value })}
                placeholder="first name"
                required
              />
              <input
                className="container__form-input"
                type="text"
                id="lastname"
                value={lastname}
                onChange={e => this.setState({ lastname: e.target.value })}
                placeholder="last name"
                required
              />
              <input
                className="container__form-input"
                type="text"
                id="othernames"
                value={othernames}
                onChange={e => this.setState({ othernames: e.target.value })
                }
                placeholder="other names"
              />
              <input
                className="container__form-input"
                type="text"
                id="username"
                value={username}
                onChange={e => this.setState({ username: e.target.value })}
                placeholder="username"
              />
              <input
                className="container__form-input"
                type="text"
                id="phonenumber"
                placeholder="phone number"
                value={phoneNumber}
                onChange={e => this.setState({ phoneNumber: e.target.value })
                }
                required
              />
              <input
                className="container__form-input"
                type="text"
                id="adminSecret"
                placeholder="admin secret key"
                value={adminSecret}
                onChange={e => this.setState({ adminSecret: e.target.value })
                }
              />
              {/* <input
                className="container__form-input"
                type="file"
                id="input-file-pic"
              /> */}
            </Fragment>
          )}
          <button className="container__form-btn" id="btn-reg" type="submit">
            {this.formType === 'signup' ? 'REGISTER' : 'LOGIN'}
          </button>
        </form>
      </div>
    );
  }
}

FormAuth.defaultProps = {
  formType: 'signup',
  isLoggedIn: false,
  fetching: false,
};

FormAuth.propTypes = {
  authorise: propTypes.func.isRequired,
  formType: propTypes.string,
  isLoggedIn: propTypes.bool,
  fetching: propTypes.bool,
  history: propTypes.oneOfType([propTypes.object, propTypes.func]).isRequired,
};

const mapStateToProps = state => ({
  isLoggedIn: state.auth.isLoggedIn,
  fetching: state.fetchStatus.fetching,

});

const FormLoginAndSignup = connect(mapStateToProps, { authorise })(FormAuth);

export default FormLoginAndSignup;
