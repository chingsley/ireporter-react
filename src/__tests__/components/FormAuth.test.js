/* eslint-disable no-multiple-empty-lines */
import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { FormAuth } from '../../components/FormAuth';

describe('Test login/signup form', () => {
  const props = {
    authorise: jest.fn(),
  };
  const mountWrapper = mount(<FormAuth {...props} />);
  it('should allow users fill the form', () => {
    expect(toJson(mountWrapper)).toMatchSnapshot();
    // console.log(mountWrapper.state());
    // const { mountWrapper } = wrap();
    // console.log(mountWrapper.find('#email').debug());
    // updateInput(form, 'input[name="email"]', 'kc.eneja@gmaiml.com');
    // updateInput(form, 'input[name="password"]', 'guessmypass');
    // expect(wrapper.state('email')).toEqual('kc.eneja@gmaiml.com');
    // expect(wrapper.state('password')).toEqual('guessmypass');
  });
  it('should allow users fill the form', () => {
    mountWrapper.find('#email').simulate('change', { target: { value: 'kc.eneja@gmail.com' } });
    mountWrapper.find('#password').simulate('change', { target: { value: 'user-password' } });
    mountWrapper.find('#firstname').simulate('change', { target: { value: 'kingsley' } });
    mountWrapper.find('#lastname').simulate('change', { target: { value: 'eneja' } });
    mountWrapper.find('#othernames').simulate('change', { target: { value: 'john' } });
    mountWrapper.find('#username').simulate('change', { target: { value: 'john' } });
    mountWrapper.find('#phonenumber').simulate('change', { target: { value: '080365603994' } });
    mountWrapper.find('#adminSecret').simulate('change', { target: { value: 'super-secret-password' } });
    expect(mountWrapper.state().email).toEqual('kc.eneja@gmail.com');
    expect(mountWrapper.state().password).toEqual('user-password');
    expect(mountWrapper.state().firstname).toEqual('kingsley');
    expect(mountWrapper.state().lastname).toEqual('eneja');
    expect(mountWrapper.state().othernames).toEqual('john');
    expect(mountWrapper.state().username).toEqual('john');
    expect(mountWrapper.state().phoneNumber).toEqual('080365603994');
    expect(mountWrapper.state().adminSecret).toEqual('super-secret-password');
  });
  it('should allow users fill the form', async () => {
    await mountWrapper.instance().onSubmit({ preventDefault: jest.fn() });
    expect(props.authorise).toHaveBeenCalled();
  });
});
