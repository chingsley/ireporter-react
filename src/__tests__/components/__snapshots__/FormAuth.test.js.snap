// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test login/signup form should allow users fill the form 1`] = `
<FormAuth
  authorise={[MockFunction]}
  fetching={false}
  formType="signup"
  isLoggedIn={false}
>
  <div
    className="container"
  >
    <form
      className="container__form"
      onSubmit={[Function]}
    >
      <input
        className="container__form-input"
        id="email"
        onChange={[Function]}
        placeholder="email"
        required={true}
        type="email"
        value=""
      />
      <input
        className="container__form-input"
        id="password"
        onChange={[Function]}
        placeholder="password"
        required={true}
        type="password"
        value=""
      />
      <input
        className="container__form-input"
        id="firstname"
        onChange={[Function]}
        placeholder="first name"
        required={true}
        type="text"
        value=""
      />
      <input
        className="container__form-input"
        id="lastname"
        onChange={[Function]}
        placeholder="last name"
        required={true}
        type="text"
        value=""
      />
      <input
        className="container__form-input"
        id="othernames"
        onChange={[Function]}
        placeholder="other names"
        type="text"
        value=""
      />
      <input
        className="container__form-input"
        id="username"
        onChange={[Function]}
        placeholder="username"
        type="text"
        value=""
      />
      <input
        className="container__form-input"
        id="phonenumber"
        onChange={[Function]}
        placeholder="phone number"
        required={true}
        type="text"
        value=""
      />
      <input
        className="container__form-input"
        id="adminSecret"
        onChange={[Function]}
        placeholder="admin secret key"
        type="password"
        value=""
      />
      <button
        className="container__form-btn"
        id="btn-reg"
        type="submit"
      >
        REGISTER
      </button>
    </form>
  </div>
</FormAuth>
`;
